### DISCUSSION TITLE
DAG dynamics

### DISCUSSION THREAD

**Discussion by dmpetrov** [treeverse/dvc#5950] [New Features & Ideas]
It would be great to have API for visual tools on top of DVC to visualize DAG status and progress - what is running, what is done, what step failed.

From a user:
https://twitter.com/wmlandau/status/1339434387738378240 Drake/targets is the library for R I was using before DVC. It follows the same concept of creating DAGs etc. They created an app to track what’s going on. This is something I would love to see in Viewer once we start having CML training jobs running in AWS.

It should be seen in a local machine as well as in remove one (run-cache might be needed).

**Comment by auvipy** [comment]
did you check apache airflow 2.0 Dag API?
